python/deltalake/table.py

```diff
@@ -210,16 +210,17 @@ def metadata(self) -> Metadata:
         """
         return self._metadata
 
-    def vacuum(self, retention_hours: int, dry_run: bool = True) -> List[str]:
+    def vacuum(self, retention_hours: Optional[int] = None, dry_run: bool = True) -> List[str]:
         """
         Run the Vacuum command on the Delta Table: list and delete files no longer referenced by the Delta table and are older than the retention threshold.
 
         :param retention_hours: the retention threshold in hours
         :param dry_run: when activated, list only the files, delete otherwise
         :return: the list of files no longer referenced by the Delta Table and are older than the retention threshold.
         """
-        if retention_hours < 0:
-            raise ValueError("The retention periods should be positive.")
+        if retention_hours:
+            if retention_hours < 0:
+                raise ValueError("The retention periods should be positive.")
```

Comment:
@fvaleye I had to get this back to python as in rust we have u64, which omits negatives.

btw, does this check looks good?

Reply:
Yes! 
Could you add a description in the docstring of the parameter `retention_hours` that explains what happens when it is None?

Reply:
sure, done